gatsby-node.js

```diff
@@ -40,3 +44,110 @@ exports.onCreateWebpackConfig = ({ stage, actions, getConfig }) => {
     actions.replaceWebpackConfig(config)
   }
 }
+
+// const checkIfDirExists = async folderPath => {
+//   try {
+//     await stat(folderPath)
+//     return true
+//   } catch (err) {
+//     if (err.code === 'ENOENT') {
+//       return false
+//     }
+//   }
+// }
+
+// const mkDirIfItDoesNotExist = async folderPath => {
+//   const doesDirExist = await checkIfDirExists(folderPath)
+//   if (!doesDirExist) {
+//     return mkdir(folderPath)
+//   }
+// }
+
+// exports.onPostBuild = async ({ graphql }) => {
+//   try {
+//     const {
+//       data: {
+//         allBlogPost: { nodes }
+//       }
+//     } = await graphql(`
+//       query GetBlogMainImages {
+//         allBlogPost {
+//           nodes {
+//             picture {
+//               gatsbyImageData(width: 850)
+//             }
+//             date
+//           }
+//         }
+//       }
+//     `)
+
+//     await mkDirIfItDoesNotExist('./public/blog/images')
+
+//     await Promise.all(
+//       nodes.map(async node => {
+//         const date = node.date.slice(0, 10)
+
+//         return mkDirIfItDoesNotExist(`./public/blog/images/${date}`)
+//       })
+//     )
+
+//     await Promise.all(
+//       nodes.map(async node => {
+//         const imagePath = node.picture.gatsbyImageData.images.fallback.src
+//         const date = node.date.slice(0, 10)
+//         return copyFile(
```

Comment:
I commented out my original solution which involved copying over the images created by `gatsby-plugin-image` to a new place in the `public` folder. At the time `copyFile` runs, gatsby hasn't created all of the images yet so you get the `can't find directory or file` error on some of the images. Maybe we could look into having Gatsby creating the images earlier, guaranteeing they are already built when we run the `copyFile` code?